consensus/spos/bls/subroundBlock.go

```diff
@@ -438,8 +494,102 @@ func (sr *subroundBlock) receivedBlockBodyAndHeader(ctx context.Context, cnsDta
 	return blockProcessedWithSuccess
 }
 
+func (sr *subroundBlock) verifyProof() bool {
+	previousAggregatedSignature, previousBitmap := sr.Header.GetPreviousAggregatedSignatureAndBitmap()
+	hasProof := len(previousAggregatedSignature) > 0 && len(previousBitmap) > 0
+	hasLeaderSignature := len(sr.Header.GetLeaderSignature()) != 0
```

Comment:
for the previous proof, to verify that the proof has the leader signature share, we should check that bit 0 is set in the bitmap (that the leader signature was considered in the aggregated signature).